src/Raven.Server/ReplicationUtil/BaseReplicationExecuter.cs

```diff
@@ -24,14 +27,16 @@ protected BaseReplicationExecuter(DocumentDatabase database)
             _log = LogManager.GetLogger(GetType());
             _database = database;
             _cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(_database.DatabaseShutdown);
+            WaitForChanges = new AsyncManualResetEvent(_cancellationTokenSource.Token);
         }
 
         public void Start()
         {
             if (_replicationThread != null)
                 return;
 
-            _replicationThread = new Thread(ExecuteReplicationLoop)
+            //haven't found better way to synchronize async method
+            _replicationThread = new Thread(() => AsyncHelpers.RunSync(ExecuteReplicationLoop))
```

Comment:
No handling of errors.
And we are running this server side, and we control the threading model here.
We can just call `.Wait()`